keras/src/models/model_test.py

```diff
@@ -232,6 +237,242 @@ def get_config(self):
         new_model = CustomModel.from_config(model.get_config())
         self.assertEqual(new_model.param, 3)
 
+    def test_reviving_functional_from_config_custom_model0(self):
+        # Functional custom model, true Functional-like constructor
+        # CustomModel.__init__ called with true functional-like args
+        # super().__init__ called with true functional-like args
+        class CustomModel0(Model):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+        inputs = layers.Input((3,))
+        outputs = layers.Dense(5)(inputs)
+        model = CustomModel0(inputs=inputs, outputs=outputs)
+
+        self.assertTrue(isinstance(model, CustomModel0))
+        self.assertTrue(isinstance(model, Functional))
+        # No way to detect that this can be serialized functionnally
+        # since the graph could have been created inside the custom
+        # __init__ with the same __init__ args.
+        config = model.get_config()
+        self.assertFalse(has_functional_config_keys(config))
```

Comment:
In all other cases of subclassing Functional, functional serialization is NOT triggered since it is not possible to detect wether the layer graph is created outside of the class (in which case functional serialization would be useful) and when the graph is created inside of __init__ in which case regular serialization that calls __init__ at the end is enough.